docs/devel/networking.md

```diff
@@ -109,3 +109,99 @@ As of now, the existent binding mechanisms are:
 - [bridge](#bridge-binding-mechanism)
 - [masquerade](#masquerade-binding-mechanism)
 - [slirp](#slirp-binding-mechanism)
+
+### Bridge binding mechanism
+Using the bridge `BindMechanism` requires a VMI configuration featuring a
+network whose interface type is `bridge` - the yaml file below can be used
+as reference, but please refer to the
+[user guide](https://kubevirt.io/user-guide/#/creation/interfaces-and-networks?id=bridge)
+for more information.
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+        - name: default
+          bridge: {}
+  networks:
+  - name: default
+    pod: {} # Stock pod network
+```
+
+Let's refer to the image below to get a better understanding of how this
+`BindMechanism` works.
+
+Bridge binding mechanism diagram
+
+![alt text](https://github.com/kubevirt/kubevirt.github.io/blob/source/assets/images/diagram.png)
+
+As can be seen in the diagram above, there are three actors at play: CNI,
+libvirt, and DHCP. For completeness sake, let's add one more actor that is
+implicit in the picture: KubeVirt.
+
+As indicated in [the introduction](#vmi-networking), the pod networking
+configuration step - performed by CNI - is out of scope of this guide. The
+focus will be instead on how KubeVirt performs bridge binding.
+
+The bridge `BindMechanism` starts off by reading the pod networking interface
+configured by CNI. It caches the assigned MAC address and the link MTU in the
+VIF structure. When CNI has configured IP address(es) on the pod networking
+interface, the bridge `BindMechanism` also caches the **first** IP address in
+the VIF, along with any routes CNI has configured.
+
+In the `preparePodNetworkInterfaces` method, KubeVirt randomizes the in-pod
+veth mac-address; the original one, will be plugged into the VM. It also
+creates an in-pod bridge, and sets the pod networking interface as a slave to
+this bridge. This happens in phase#1 - i.e. performed by virt-handler on
+behalf of virt-launcher.
+
+The `preparePodNetworkInterfaces` method performs one other operation: **if**
+the pod networking interface featured any IP address, it will delete the first
+one; remember this address is cached in the VIF structure It will be needed
+for phase#2.
+
+In phase#2 (executed by virt-launcher, unprivileged) the domain xml which will
+create the VMI is generated. It first creates a rough domxml of each interface
+of the VM, whose sole purpose is to instruct how to connect to the in-pod
+bridge. The generated interface xml element looks like:
+
+```xml
+ <interface type='bridge'>
+    <source bridge='k6t-eth0'/>
+    <model type='virtio'/>
+ </interface>
+```
+
+Afterwards, but still in phase#2, KubeVirt will decorate the aforementioned
+interface xml, specifying the link MTU - and optionally the MAC address - via
```

Comment:
How come the `mac` is optional?

Reply:
It generates one when the mac is not specified.

Reply:
I expected that the mac address of the pod interface to be forwarded to the VM.
Isn't this the objective?

Reply:
You're totally right, I was mixing concepts when I replied "generates one when the mac is not specified". That's not true.

A mac address can be requested on the VMI spec, as per the following snippet:
```yaml
---
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachineInstance
metadata:
  ...
  name: vmi-masquerade
spec:
  domain:
    devices:
      ...
      interfaces:
      - masquerade: {}
        name: testmasquerade
        macAddress: AA:BB:CC:DD:EE:FF
...
```

When the MAC is not specified in the interface spec, it **does** "move" the mac from the pod interface to the VM, as you've pointed out.

Good catch.